AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -273,6 +377,23 @@ - (void)reloadDataWithAnimationOptions:(ASDataControllerAnimationOptions)animati
   }];
 }
 
+#pragma mark - Initial & Reload Data Hooks (Subclass API)
+
+- (void)willPerformInitialDataLoading
+{
+  // Implemented by subclasses
```

Comment:
Maybe // Optional hook for subclasses (might be more clear that ASDataController is a concrete class, does not require subclassing & implementation of these methods)